lib/src/lifecycle_module.dart

```diff
@@ -510,15 +514,8 @@ abstract class LifecycleModule extends SimpleModule with Disposable {
           onError: (error, stackTrace) =>
               _didUnloadChildModuleController.addError);
 
-      // The child module may not reach an unloaded state successfully, but
-      // should always eventually be disposed. For this reason, we listen for
-      // its disposal before removing it from the list of child modules.
-      // ignore: unawaited_futures
-      childModule.didDispose.then((_) {
-        _childModules.remove(childModule);
-      });
```

Comment:
I think we need to keep this in place so that unloaded child modules aren't retained in `_childModules` until the parent is unloaded (not sure why we don't have a test for that 😞). 

If we restore this, we should be able to remove the `_childModules.clear()` disposer, but I wouldn't be opposed to keeping it around to be safe.